process/block/shardblock.go

```diff
@@ -250,6 +254,44 @@ func (sp *shardProcessor) ProcessBlock(
 	return nil
 }
 
+func (sp *shardProcessor) setShardConsensusData(headerHandler data.HeaderHandler) error {
+	// give transaction coordinator the consensus group validators addresses where to send the rewards.
+	consensusAddresses, err := sp.nodesCoordinator.GetValidatorsRewardsAddresses(
+		headerHandler.GetPrevRandSeed(),
+		headerHandler.GetRound(),
+		sp.shardCoordinator.SelfId(),
+	)
+	if err != nil {
+		return err
+	}
+
+	sp.SetConsensusData(consensusAddresses, headerHandler.GetRound())
+
+	return nil
+}
+
+func (sp *shardProcessor) setMetaConsensusData(finalizedMetaBlocks []data.HeaderHandler) error {
```

Comment:
this functions should be in a separate component.

Reply:
the logic for selecting the right addresses etc was moved to the address handler, these simply call the interface, so there is no need of creating a separate component.

Reply:
There is still logic inside the function, which could have its own component

Reply:
We should thrive to make blockprocessor smaller and to call interfaces.

Reply:
The processor should be refactored in a separate task.
There is no need of a component that only passes data from one component to the other, it makes no sense.